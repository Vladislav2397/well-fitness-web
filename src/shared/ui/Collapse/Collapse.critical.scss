.collapse {
    border-bottom: 1px solid #ccc;

    &__header {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
    }

    &__icon {
        width: toRem(30);
        height: toRem(30);
        display: flex;
        justify-content: center;
        align-items: center;
    }
}
