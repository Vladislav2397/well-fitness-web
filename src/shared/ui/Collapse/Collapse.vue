<template lang="pug">

.b-collapse
    .__header(
        @click="onClick"
    )
        slot(
            name="header"
        )
        .__icon V
    .__content(
        v-if="isOpen"
    )
        slot

</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

export type CollapseProps = {
    //
}

@Component
export default class Collapse extends Vue {
    isOpen = false

    onClick() {
        this.isOpen = !this.isOpen
    }
}
</script>

<!-- <style lang="scss" src="./Collapse.critical.scss" /> -->
<!-- <style lang="scss" src="./Collapse.main.scss" /> -->
