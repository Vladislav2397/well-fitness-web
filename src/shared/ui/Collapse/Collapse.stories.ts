import { Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import Collapse, { type CollapseProps } from './Collapse.vue'
import { defineControls } from '@/shared/lib/stories'

export default {
    title: 'undefined-path/Collapse',
    component: Collapse,
    argTypes: defineControls({
        //
    }),
}

const Template: Story<CollapseProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { Collapse },
    methods: {
        handle: action('clicked'),
    },
    template: `<Collapse v-bind="$props" @click="handle" />`,
})

export const Default = Template.bind({})
Default.args = {
    //
}
