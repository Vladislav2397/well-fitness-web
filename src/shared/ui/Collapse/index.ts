import Collapse, { type CollapseProps } from './Collapse.vue'

export { Collapse, CollapseProps }
