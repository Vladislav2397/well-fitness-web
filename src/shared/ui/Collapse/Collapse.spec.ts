import { shallowMount } from '@vue/test-utils'
import Collapse /*, { type CollapseProps }*/ from './Collapse.vue'

describe('test Collapse', () => {
    test('snapshot', () => {
        const wrapper = shallowMount(Collapse, {
            propsData: {
                //
            } /* as CollapseProps*/,
        })
        expect(wrapper).toMatchSnapshot()
    })
})
