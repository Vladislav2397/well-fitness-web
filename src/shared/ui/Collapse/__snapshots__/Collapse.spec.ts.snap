// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Collapse snapshot 1`] = `
<div class="b-collapse">
  <div class="__header">
    <div class="__title"></div>
    <div class="__icon">V</div>
  </div>
  <!---->
</div>
`;
